pages/api/gifts/[id].ts

```diff
@@ -35,37 +41,22 @@ export default async function handler(
 }
 
 async function handleGET(req: NextApiRequest, res: NextApiResponse) {
-  if (typeof req.query.id !== 'string') {
-    throw new Error('Invalid ID', { cause: 'idError' });
-  }
-  const giftRequest = await axios.get(`${baseURL}/${req.query.id}`);
+  const giftRequest = await axios.get(`${baseURL}/${queryID}`);
   return res.status(giftRequest.status).send(giftRequest.data as Gift);
 }
 
 async function handlePATCH(req: NextApiRequest, res: NextApiResponse) {
-  if (typeof req.query.id !== 'string') {
-    throw new Error('Invalid ID', { cause: 'idError' });
-  }
-  const patchRequest = await axios.patch(
-    `${baseURL}/${req.query.id}`,
-    req.body,
-  );
+  const patchRequest = await axios.patch(`${baseURL}/${queryID}`, req.body);
   return res.status(patchRequest.status).json(req.body);
 }
 
 async function handlePUT(req: NextApiRequest, res: NextApiResponse) {
-  if (typeof req.query.id !== 'string') {
-    throw new Error('Invalid ID', { cause: 'idError' });
-  }
-  const putRequest = await axios.put(`${baseURL}/${req.query.id}`, req.body);
+  const putRequest = await axios.put(`${baseURL}/${queryID}`, req.body);
   return res.status(putRequest.status).json(req.body);
 }
 
 async function handleDELETE(req: NextApiRequest, res: NextApiResponse) {
-  if (typeof req.query.id !== 'string') {
-    throw new Error('Invalid ID', { cause: 'idError' });
-  }
-  const deleteRequest = await axios.delete(`${baseURL}/${req.query.id}`);
+  const deleteRequest = await axios.delete(`${baseURL}/${queryID}`);
   return res.status(deleteRequest.status).json(req.body);
```

Comment:
Delete ei yleensä palauta bodya, joten olisiko mennyt näin:
```suggestion
  return res.status(deleteRequest.status).send();
```